doc/design/networking/Investigation-on-NSX-integration.md

```diff
@@ -0,0 +1,254 @@
+
+
+## A Working Installation of NSX-v
+For VMware engineers using Nimbus, here is a document about how to install and configure NSX with vSphere: How to set up VM-to-VM communication on NSX logical switch with Nimbus
+https://confluence.eng.vmware.com/pages/viewpage.action?pageId=209964551
+
+## About NSX Security Groups and Security Policies
+This section is a brief introduction about NSX security groups and security policies (some words are taken from this document http://www.vmware.com/content/dam/digitalmarketing/vmware/en/pdf/products/nsx/vmw-nsx-network-virtualization-design-guide.pdf).  
+  * Security Groups
+
+    NSX provides various grouping mechanisms :
+      * vCenter Objects: VMs, Distributed Switches, Clusters, etc.
+      * VM Properties: vNICs, VM names, VM operating Systems, etc.
+      * NSX Objects: Logical Switches, Security Tags, Logical Routers, etc. 
+    In this document, we propose to use vNICs as the grouping criteria. We will discuss more on this later.
+
+  * Security Policies
+
+    NSX provides security policies as a way to group rules for security controls that will be applied to one or more security groups.
+    Each security policy contains the following :
+      * Firewall rules: 
+        * NSX built in distributed firewall.
+      * weight
+        * Weight of a policy determines the rank of the policy versus other policies in the NSX eco-system. Higher weight rules are executed first.
+
+## An Overview of Workflow
+
+Current VIC implementation may have the container bridge networks all reside on a single port group and only be separated by IP space. Our container networks (created using `docker network create`) are not isolated from each other. For example, a containerVM in one container network can change its IP and be able to reach containerVMs in another container network.
+
+Our research shows that we can use NSX security groups and policies on top of current VIC implementation and achieve micro-segmentation. 
+
+Our goals :
+  * VMs can communicate with VMs in the same network (security group).
+  * VMs cannot talk to VMs in a differrent network (security group).
+  * one VM to be added to more than one network (security group).
+
+In the following example, we see that : 
+  * containerVM1 and containerVM2 are in default bridge network
+  * containerVM3 , containerVM4 and containerVM5 are in bridge network `network-a` 
+  * containerVM5 and containerVM6 are in bridge network `network-b`
+    (note that adding containerVM5 to both network `network-a` and network `network-b` are not supported on current VIC yet.)
+
+![An Example of How We Apply Security Group on Container Networks](pics/example-SG-vs-network.png)
+
+Let us use this example to explain the workflow.
+
+### Prerequisite
+We assume vSphere admin already has NSX installed and configured on vCenter. That means there is an NSX logical switch created by the admin, which automatically creates a port group on distributed switch.
```

Comment:
Yes. We can do that. As long as NSX has been installed, in the installer we can make API calls to NSX to create logical switch, which will create corresponding port group :)